packages/editor-ui/src/utils/typeGuards.ts

```diff
@@ -98,3 +111,37 @@ export function isRouteLocationRaw(value: unknown): value is RouteLocationRaw {
 export function isComponentPublicInstance(value: unknown): value is ComponentPublicInstance {
 	return value !== null && typeof value === 'object' && '$props' in value;
 }
+
+export function isWorkflowResource(value: Resource): value is WorkflowResource {
+	return value.resourceType === 'workflows';
+}
+
+export function isFolderResource(value: Resource): value is FolderResource {
+	return value.resourceType === 'folders';
+}
+
+export function isVariableResource(value: Resource): value is VariableResource {
+	return value.resourceType === 'variables';
+}
+
+export function isCredentialsResource(value: Resource): value is CredentialsResource {
+	return value.resourceType === 'credentials';
+}
+
+export function isSharedResource(
+	value: Resource,
+): value is WorkflowResource | FolderResource | CredentialsResource {
+	return isWorkflowResource(value) || isFolderResource(value) || isCredentialsResource(value);
+}
+
+export function isResourceSortableByDate(
+	value: Resource,
+): value is WorkflowResource | FolderResource | CredentialsResource {
+	return isWorkflowResource(value) || isFolderResource(value) || isCredentialsResource(value);
+}
+
+export function isResponseWorkflowResource(
+	value: WorkflowListResourceDB,
+): value is WorkflowResourceDB {
```

Comment:
this typecheck is not adding anything since it's the same type

Reply:
Good catch